x-pack/plugins/rollup/__jest__/integration_tests/job_create.test.js

```diff
@@ -0,0 +1,1298 @@
+      await form.setInputValue('rollupIndexPattern', JOB_TO_CREATE.indexPattern, true);
+      form.setInputValue('rollupIndexName', JOB_TO_CREATE.rollupIndex);
+      break;
+    case 'date-histogram':
+      form.setInputValue('rollupJobInterval', JOB_TO_CREATE.interval);
+      break;
+    default:
+      return;
+  }
+};
+
+const initGoToStep = (fillFormFields, clickNextStep) => async (step) => {
+  if (!step || step === 1) {
+    return;
+  }
+
+  await fillFormFields('logistics');
+  clickNextStep();
+
+  if (step > 2) {
+    await fillFormFields('date-histogram');
+    clickNextStep();
+  }
+
+  if (step > 3) {
+    clickNextStep();
+  }
+
+  if (step > 4) {
+    clickNextStep();
+  }
+
+  if (step > 5) {
+    clickNextStep();
+  }
+};
+
+const initTestBed = () => {
+  const testBed = registerTestBed(JobCreate, {}, rollupJobsStore)();
+  const userActions = initUserActions(testBed.component, testBed.findTestSubject);
+  const fillFormFields = initFillFormFields(testBed.form);
+  const goToStep = initGoToStep(fillFormFields, userActions.clickNextStep);
+  const getEuiStepsHorizontalActive = () => testBed.component.find('.euiStepHorizontal-isSelected').text();
+
+  return {
+    ...testBed,
+    userActions: {
+      ...userActions
+    },
+    form: {
+      ...testBed.form,
+      fillFormFields,
+    },
+    goToStep,
+    getEuiStepsHorizontalActive,
+  };
+};
+
+const nextTick = async () => new Promise((resolve) => setTimeout(resolve));
+
+const mockServerResponses = server => {
+  const mockIndexPatternValidityResponse = (response) => {
+    const defaultResponse = {
+      doesMatchIndices: true,
+      doesMatchRollupIndices: false,
+      dateFields: ['foo', 'bar'],
+      numericFields: [],
+      keywordFields: [],
+    };
+    server.respondWith(/\/api\/rollup\/index_pattern_validity\/.*/, [
+      200,
+      { 'Content-Type': 'application/json' },
+      JSON.stringify({ ...defaultResponse, ...response }),
+    ]);
+  };
+
+  const mockCreateJob = () => {
+    server.respondWith(/\/api\/rollup\/create/, [
+      200,
+      { 'Content-Type': 'application/json' },
+      JSON.stringify({}),
+    ]);
+  };
+
+  mockIndexPatternValidityResponse();
+  mockCreateJob();
+
+  return { mockIndexPatternValidityResponse };
+};
+
+describe('Create Rollup Job', () => {
+  let server;
+  let findTestSubject;
+  let testSubjectExists;
+  let userActions;
+  let getFormErrorsMessages;
+  let getMetadataFromEuiTable;
+  let form;
+  let mockIndexPatternValidityResponse;
+  let getEuiStepsHorizontalActive;
+  let goToStep;
+
+  beforeEach(() => {
+    server = sinon.fakeServer.create();
+    server.respondImmediately = true;
+    ({ mockIndexPatternValidityResponse } = mockServerResponses(server));
+    ({
+      findTestSubject,
+      testSubjectExists,
+      userActions,
+      getFormErrorsMessages,
+      getMetadataFromEuiTable,
+      form,
+      goToStep,
+      getEuiStepsHorizontalActive,
+    } = initTestBed());
```

Comment:
Seeing this, I feel like it really begs to be refactored into the [page object model](https://medium.com/tech-tajawal/page-object-model-pom-design-pattern-f9588630800b). Then instead of having `findTestSubject` calls everywhere, like this:

```js
expect(findTestSubject('rollupJobNextButton').props().disabled).toBe(true);
```

The page object could expose individual parts of the UI, so the code reads a bit more like a story:

```js
expect(rollupJobWizard.nextButton.props().disabled).toBe(true);
```

This would also mean `testSubjectExists` is no longer necessary, since we could do this instead:

```js
expect(rollupJobWizard.nextButton.exists()).toBe(true);
```

Reply:
I will have to give it some thought as it will complicate things and I'm not sure it brings _that_ much value. For a few objects always present on the screen, it might be nice. But for elements that only appear in the DOM on a specific step (or when the detail panel opens) then it either does not work, or we have to add special methods to dynamically update the objects (the complication I'm talking about).

And then we'd have 2 lines (1 to update the objects and 1 to test) instead of one. We can talk about it over zoom if u want.

Reply:
Cool, I'm also willing to take a stab at it myself in a separate PR.